### PR TITLE
Closes #394 | Implement dataloader for UCLA Phonetic Corpus

### LINKED ISSUE #394: Create dataset loader for UCLA Phonetic Corpus
Dataloader name: `ucla_phonetic/ucla_phonetic.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?ucla_phonetic

|     Dataset| ucla_phonetic |
|-------------|---|
| Description | This dataset contains audio recordings and phonetic transcriptions of word utterances for various low-resource SEA languages. Each language has a directory of text and audio files, with the latter forming one data subset.  The dataset is prepared from the online UCLA phonetic dataset, which contains 7000 utterances across 100 low-resource languages, phonetically aligned using various automatic approaches, and manually fixed for misalignments. |
| Subsets | - |
| Languages | ace, brv, hil, hni, ilo, khm, mak, mya, pam |
| Tasks | Automatic Speech Recognition |
| License | Creative Commons Attribution Non Commercial 2.0 (cc-by-nc-2.0) |
| Homepage | https://github.com/xinjli/ucla-phonetic-corpus?tab=readme-ov-file |
| HF URL | - |
| Paper URL | https://www.cs.cmu.edu/~awb/papers/ICASSP21_Multilingual_Phonetic_Dataset.pdf |

### PR DESCRIPTION
**Pull request by ssun32**
closes #394 

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/ucla_phonetic/ucla_phonetic.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+_DATASETNAME = "ucla_phonetic"
+
+_DESCRIPTION = """\
+This dataset contains audio recordings and phonetic transcriptions of word utterances for various low-resource SEA languages.
+Each language has a directory of text and audio files, with the latter forming one data subset.
+The dataset is prepared from the online UCLA phonetic dataset, which contains 7000 utterances across 100 low-resource languages, phonetically aligned using various automatic approaches, and manually fixed for misalignments.
+"""
+
+_HOMEPAGE = "https://github.com/xinjli/ucla-phonetic-corpus?tab=readme-ov-file"
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/ucla_phonetic/ucla_phonetic.py:50]:
```suggestion
_HOMEPAGE = "https://github.com/xinjli/ucla-phonetic-corpus"
```

File: seacrowd/sea_datasets/ucla_phonetic/ucla_phonetic.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                description="UCLA Phonetic seacrowd+sptext for ace",
+                schema="seacrowd_sptext",
+                subset_id="ace_seacrowd_sptext",
+            ),
+        ]
+        + [seacrowd_config_constructor(lang, "source", _SOURCE_VERSION) for lang in _LANGUAGES]
+        + [seacrowd_config_constructor(lang, "seacrowd_sptext", _SEACROWD_VERSION) for lang in _LANGUAGES]
+    )
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_ace_source"
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/ucla_phonetic/ucla_phonetic.py:109]: What is `ace` again? If it's part of the dataset name, maybe we should put it into `_DATASETNAME`

**Review by ljvmiranda921 — COMMENTED** [review]
Hi! Just have a few nits

* Add an `__init__.py` file
* Can you rerun `make check_file=...` again?

Thank you!

File: seacrowd/sea_datasets/ucla_phonetic/ucla_phonetic.py (hunk 1 above)
Comment (ssun32) [reply seacrowd/sea_datasets/ucla_phonetic/ucla_phonetic.py]: added to new commit

File: seacrowd/sea_datasets/ucla_phonetic/ucla_phonetic.py (hunk 2 above)
Comment (fhudi) [reply seacrowd/sea_datasets/ucla_phonetic/ucla_phonetic.py]: @ljvmiranda921 I think `ace` refers to `_LANGUAGES[0]`.
Sorry for suddenly butting in, saw the incoming email and thought this is kind of obvious, but CMIIW.

**Review by ljvmiranda921 — APPROVED** [review]
LGTM! Let's wait for @yongzx 's review

**Comment by yongzx** [comment]
LGTM, and tested with test suite. Merging now.

**Review by yongzx — APPROVED** [review]

**Pull request merged**